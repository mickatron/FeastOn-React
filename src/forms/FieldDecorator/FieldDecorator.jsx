import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import classNameType from 'js/shared/proptypes/className';


export default class FieldDecorator extends React.PureComponent  {
  
  static propTypes = {
    className: classNameType,
    children: PropTypes.node.isRequired,
    disabled: PropTypes.bool,
    readOnly: PropTypes.bool,
    required: PropTypes.bool,
    isValid: PropTypes.shape({
      valid: PropTypes.bool,
      warning: PropTypes.bool,
      info: PropTypes.bool,
    }),
  };
  static defaultProps = {
    isValid: {
      valid: undefined,
      warning: undefined,
      info: undefined,
    },
  };
  
  render() {
    const { className, 
      children, 
      disabled, 
      readOnly, 
      required, 
      style,
      isValid, 
    } = this.props;

    return (
      <div style={style} className={ classnames(className, {
        'f-FieldDecorator--disabled': disabled,
        'f-FieldDecorator--readOnly': readOnly,
        'f-FieldDecorator--required': required,
        'f-FieldDecorator--valid': isValid.valid === true,
        'f-FieldDecorator--invalid': isValid.valid === false,
        'f-FieldDecorator--warning': !isValid.warning,
        'f-FieldDecorator--info': !isValid.info,
      })}>
        {children}
      </div>
    );
  };
}