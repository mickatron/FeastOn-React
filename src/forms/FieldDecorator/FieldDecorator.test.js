import { shallow } from 'enzyme';
import React from 'react';

import FieldDecorator from './FieldDecorator';
// import data from '../mock/data'

describe('FieldDecorator', () => { 
  let component;
  beforeAll(() => {
    component = shallow(
      <FieldDecorator 
        className="test"
      />
    );
  });

  describe('renders', () => { 
    it('has passed in className', ()=>{
      expect(component).toMatchSnapshot();
      expect(component.find('.test').exists()).toBeTruthy();
    });
  });
});