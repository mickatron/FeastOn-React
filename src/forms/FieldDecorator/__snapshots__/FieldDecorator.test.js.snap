// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldDecorator renders has passed in className 1`] = `
<div
  className="test f-FieldDecorator--warning f-FieldDecorator--info"
/>
`;
